test/unit/test_rnn.py

```diff
@@ -77,3 +77,96 @@ def test_get_stacked_rnn(config, expected_cell):
         cell = cell.base_cell
     assert isinstance(cell, expected_cell)
     assert cell._num_hidden, config.num_hidden
+
+def test_residual_cell_parallel_input():
+    num_hidden = 128
+    batch_size = 256
+    parallel_size = 64
+
+    input_shape = (batch_size, num_hidden)
+    states_shape = (batch_size, num_hidden)
+    parallel_shape = (batch_size, parallel_size)
+
+    input = mx.sym.Variable("input")
```

Comment:
fine with me if the code checkers don't complain shadowing a reserved python word. I usually go for 'inp' :D